### DISCUSSION TITLE
define_table() throws an error for name column

### DISCUSSION THREAD

**Discussion by thee-cos** [osm2pgsql-dev/osm2pgsql#2267] [Flex Recipes]
I am wondering if I am doing something wrong or if this is a legitimate issue with the code. I am trying to define my own table for insertion into the database. 
I have the following versions of software:
osm2pgsql version 2.0.0
Database version: 16.3
PostGIS version: 3.4

the table definition in question:
```
local pois = osm2pgsql.define_table({
    name = 'pois',
    ids = { type = 'node', id_column = 'id', type_column = 'obj_type' },
    columns = {
        { column = 'name',      type = 'text' },
	    { column = 'tags',      type = 'jsonb' },
	    { column = 'geom',      type = 'point', projection = srid, not_null = true },
    }
})
```

Error Response:

> ERROR: Error loading lua config: [string "..."]:14: Error in 'define_table': The table must contain a 'name' string field

I get the same error for the example definition in the manual.
```
local restaurants = osm2pgsql.define_table({
    name = 'restaurants',
    ids = { type = 'node', id_column = 'node_id' },
    columns = {
        { column = 'name', type = 'text' },
        { column = 'tags', type = 'jsonb' },
        { column = 'geom', type = 'point' }
}})

```

**Comment by joto** [comment]
Your example works fine here. Please show us your complete command line and the whole config file.

**Reply by thee-cos** [reply]
`osm2pgsql -c -d test -s -O flex --style=./osm/lua_config/custom_v2.lua ./osm/milan_bbox.pbf`

The .lua file type is unsupported for upload for some reason

**Reply by joto** [reply]
There must be something before the part of the config file you quoted that breaks things. Try trimming down your Lua file to identify the problem. If you can't find it copy-and-paste the config file in a comment here.

**Reply by thee-cos** [reply]
It was a problem with a different define_way_table with the name located inside the table. My mistake
